Build failure: virtualbox

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

- build *virtualbox*

### Can Hydra reproduce this build failure?

The last build job was manually cancelled.

### Link to Hydra build job

https://hydra.nixos.org/build/309205708

### Relevant log output

```console
In file included from /build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:82:
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp: In function 'int rtHttpUpdateAutomaticProxyDisable(PRTHTTPINTERNAL)':
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:702:27: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
  702 |     pThis->enmProxyType = CURLPROXY_HTTP;
      |                           ^~~~~~~~~~~~~~
      |                           |
      |                           long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp: In function 'int rtHttpConfigureProxyFromUrl(PRTHTTPINTERNAL, const char*)':
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:877:33: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
  877 |                 enmProxyType  = CURLPROXY_HTTP;
      |                                 ^~~~~~~~~~~~~~
      |                                 |
      |                                 long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:885:33: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
  885 |                 enmProxyType  = CURLPROXY_HTTPS;
      |                                 ^~~~~~~~~~~~~~~
      |                                 |
      |                                 long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:893:32: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
  893 |                 enmProxyType = CURLPROXY_SOCKS4;
      |                                ^~~~~~~~~~~~~~~~
      |                                |
      |                                long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:895:32: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
  895 |                 enmProxyType = CURLPROXY_SOCKS4A;
      |                                ^~~~~~~~~~~~~~~~~
      |                                |
      |                                long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:897:32: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
  897 |                 enmProxyType = CURLPROXY_SOCKS5;
      |                                ^~~~~~~~~~~~~~~~
      |                                |
      |                                long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:899:32: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
  899 |                 enmProxyType = CURLPROXY_SOCKS5_HOSTNAME;
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                                |
      |                                long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:903:32: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
  903 |                 enmProxyType = CURLPROXY_HTTP;
      |                                ^~~~~~~~~~~~~~
      |                                |
      |                                long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp: In function 'int RTHttpSetProxy(RTHTTP, const char*, uint32_t, const char*, const char*)':
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:2097:43: error: invalid conversion from 'long int' to 'curl_proxytype' [-fpermissive]
 2097 |     return rtHttpUpdateProxyConfig(pThis, CURLPROXY_HTTP, pcszProxy, uPort ? uPort : 1080, pcszProxyUser, pcszProxyPwd);
      |                                           ^~~~~~~~~~~~~~
      |                                           |
      |                                           long int
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp:594:74: note:   initializing argument 2 of 'int rtHttpUpdateProxyConfig(PRTHTTPINTERNAL, cur>
  594 | static int rtHttpUpdateProxyConfig(PRTHTTPINTERNAL pThis, curl_proxytype enmProxyType, const char *pszHost,
      |                                                           ~~~~~~~~~~~~~~~^~~~~~~~~~~~
/build/VirtualBox-7.2.0/src/VBox/Runtime/generic/http-curl.cpp: In function 'int RTHttpQueryProxyInfoForUrl(RTHTTP, const char*, PRTHTTPPROXYINFO)':
```

### Additional context

_No response_

### System metadata

fetching path input 'path:/nix/store/ybmnblw90230yl4p0l18ghwx9ry597bz-source'
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.50, NixOS, 25.11 (Xantusia), 25.11.20251005.d7f52a7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/misile/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/misile/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/misile/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/misile/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/nk1i09ppsgs803sy63dx86m4c4kpsfph-lix-2.93.3/share`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/ybmnblw90230yl4p0l18ghwx9ry597bz-source`

### Notify maintainers

@svanderburg @FriedrichAltheide @blitz

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/virtualization/virtualbox/default.nix
+++ b/pkgs/applications/virtualization/virtualbox/default.nix
@@ -260,6 +260,11 @@ stdenv.mkDerivation (finalAttrs: {
       ./qt-dependency-paths.patch
       # https://github.com/NixOS/nixpkgs/issues/123851
       ./fix-audio-driver-loading.patch
+      # curl 8.16 upgrade breakage
+      (fetchpatch {
+        url = "https://salsa.debian.org/pkg-virtualbox-team/virtualbox/-/raw/dbf9a6ef75380ebd2705df0198c6ac8073d0b4cb/debian/patches/new-curl.patch";
+        hash = "sha256-WWnCWdXlJo9jTr8yXA0NxcDQBScryuu/53wyX0rhszk=";
+      })
     ];
 
   postPatch = ''

